src/components/date-range-picker/predefined-ranges-area.ts

```diff
@@ -0,0 +1,119 @@
+import { LitElement, html } from 'lit-element';
+import { property } from 'lit/decorators.js';
+import { CalendarDay } from '../calendar/model.js';
+import IgcChipComponent from '../chip/chip.js';
+import { registerComponent } from '../common/definitions/register.js';
+import {
+  type IgcDateRangePickerResourceStrings,
+  IgcDateRangePickerResourceStringsEN,
+} from '../common/i18n/date-range-picker.resources.js';
+import type { CustomDateRange } from './date-range-picker.js';
+
+/**
+ * The predefined ranges area component is used within the `igc-date-range picker` element and it
+ * displays a set of chips with predefined date ranges. The component allows users to quickly select
+ * a predefined date range value. Users can also provide custom ranges to be displayed as chips.
+ *
+ * @element igc-predefined-ranges-area
+ */
+export default class IgcPredefinedRangesAreaComponent extends LitElement {
+  public static readonly tagName = 'igc-predefined-ranges-area';
+
+  /* blazorSuppress */
+  public static register() {
+    registerComponent(IgcPredefinedRangesAreaComponent, IgcChipComponent);
+  }
+
+  /**
+   * Whether the control will show chips with predefined ranges.
+   * @attr
+   */
+  @property({
+    reflect: true,
+    type: Boolean,
+    attribute: 'use-predefined-ranges',
+  })
+  public usePredefinedRanges = false;
+
+  /**
+   * Renders chips with custom ranges based on the elements of the array.
+   */
+  @property({ type: Array })
+  public customRanges: CustomDateRange[] = [];
+
+  /** The resource strings of the date range area component. */
+  @property({ attribute: false })
+  public resourceStrings: IgcDateRangePickerResourceStrings =
+    IgcDateRangePickerResourceStringsEN;
+
+  private _predefinedRanges: CustomDateRange[] = [
```

Comment:
I see that you have tests which cover setting a specific localization property
and verifying it shows correctly here. That is fine as long as the `resourceStrings` reference is the same. Have you tested what happens when a developer passes a whole new resource object to the date range picker and how that reflects in the predefined ranges?